pymc3/tests/test_distributions_random.py

```diff
@@ -492,43 +421,316 @@ class TestMoyal(BaseTestCases.BaseTestCase):
+            params["expected_dist_outcome"] = params["expected_dist"](
+                **params["expected_dist_params"], size=params.get("size", 15)
+            )
+        for test in tests_to_run:
+            test(**params)
 
-@pytest.mark.skip(reason="This test is covered by Aesara")
-class TestDirichlet(SeededTest):
-    @pytest.mark.parametrize(
-        "shape, size",
-        [
-            ((2), (1)),
-            ((2), (2)),
-            ((2, 2), (2, 100)),
-            ((3, 4), (3, 4)),
-            ((3, 4), (3, 4, 100)),
-            ((3, 4), (100)),
-            ((3, 4), (1)),
-        ],
-    )
-    def test_dirichlet_random_shape(self, shape, size):
-        out_shape = to_tuple(size) + to_tuple(shape)
-        assert pm.Dirichlet.dist(a=np.ones(shape)).random(size=size).shape == out_shape
+    def _check_pymc_dist_matches_expected(
+        self,
+        pymc_dist_output: TensorVariable,
+        expected_dist_outcome: np.ndarray,
+        decimal: int = 6,
+        **kwargs,
+    ) -> None:
+        assert_array_almost_equal(pymc_dist_output.eval(), expected_dist_outcome, decimal=decimal)
+
+    def _check_pymc_params_match_rv_op(
+        self,
+        pymc_dist_output: TensorVariable,
+        expected_rv_op_params: Dict[str, Any],
+        decimal: int = 6,
+        **kwargs,
+    ) -> None:
+        aesera_dist_inputs = self._get_inputs_from_apply_node_outputs(pymc_dist_output)[3:]
+        assert len(expected_rv_op_params) == len(aesera_dist_inputs)
+        for (expected_name, expected_value), actual_variable in zip(
+            expected_rv_op_params.items(), aesera_dist_inputs
+        ):
+            assert_almost_equal(expected_value, actual_variable.eval(), decimal=decimal)
+
+    def _get_inputs_from_apply_node_outputs(self, outputs: TensorVariable) -> List[Any]:
+        parents = outputs.get_parents()
+        if not parents:
+            raise Exception("Parent Apply node missing for output")
+        # I am assuming there will always only be 1 Apply parent node in this context
+        return parents[0].inputs
+
+    def _get_numpy_distribution(self, dist_name: str) -> Callable:
+        return functools.partial(getattr(np.random.RandomState, dist_name), self.get_random_state())
+
+    def _get_scipy_distribution(self, dist_name: str) -> Callable:
+        return functools.partial(getattr(st, dist_name).rvs, random_state=self.get_random_state())
+
+    def test_gumbel(self):
+        params = {
+            "pymc_dist": pm.Gumbel,
+            "pymc_dist_params": {"mu": 1.5, "beta": 3.0},
+            "expected_rv_op_params": {"mu": 1.5, "beta": 3.0},
+            "expected_dist": self._get_scipy_distribution("gumbel_r"),
+            "expected_dist_params": {"loc": 1.5, "scale": 3.0},
+            "size": 15,
+        }
+        self._run_distribution_checks(
+            params, [self._check_pymc_params_match_rv_op, self._check_pymc_dist_matches_expected]
+        )
+
+    def test_normal(self):
```

Comment:
We should add tests for `tau` and `sd` parametrizations as well. Same for the HalfNormal